cranelift-codegen/meta/src/cdsl/type_inference.rs

```diff
@@ -161,8 +161,8 @@ impl TypeEnvironment {
 
         // Check extra conditions for InTypeset constraints.
         if let Constraint::InTypeset(tv, _) = &constraint {
-            assert!(tv.base.is_none());
-            assert!(tv.name.starts_with("typeof_"));
+            assert!(tv.base.is_none(), "tv.base == {:?}", tv.base);
```

Comment:
It's useful to have an explicit diagnostic message here. Can you tweak it to `"type variable is {:?}, while expecting none"`?

Reply:
```suggestion
        assert!(tv.base.is_none(), "type variable is {:?}, while expecting none", tv);
```